var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -34,65 +39,105 @@ class Mesa(AutotoolsPackage):
     list_url = "https://mesa.freedesktop.org/archive"
     list_depth = 2
 
+    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d')
+    version('17.1.4', 'be2ef7c9edec23b07f74f6512a6a6fa5')
     version('17.1.3', '1946a93d543bc219427e2bebe2ac4752')
+    version('17.1.1', 'a4844bc6052578574f9629458bcbb749')
     version('13.0.6', '1e5a769bc6cfd839cf3febcb179c27cc')
     version('12.0.6', '1a3d4fea0656c208db59289e4ed33b3f')
     version('12.0.3', '1113699c714042d8c4df4766be8c57d8')
 
-    variant('swrender', default=False,
-        description="Build with (gallium) software rendering.")
-
+    variant('swrender', default=True,
+            description="Build with (gallium) software rendering.")
+    variant('hwrender', default=False,
+            description="Build with (DRI) hardware rendering.")
     variant('llvm', default=False,
-        description="Use llvm for rendering pipes.")
+            description="Use llvm for rendering pipes.")
 
     # General dependencies
-    depends_on('python@2.6.4:')
-    depends_on('py-mako@0.3.4:', type=('build', 'run'))
+    depends_on('pkg-config@0.9.0:', type='build')
     depends_on('flex@2.5.35:', type='build')
     depends_on('bison@2.4.1:', type='build')
-    depends_on('pkg-config@0.9.0:', type='build')
-
-    # Off-screen with llvmpipe
-    # Note: there must be a better way of selecting the preferred llvm
-    depends_on('llvm+link_dylib', when='+llvm')
-
-    # For DRI and hardware acceleration
+    depends_on('binutils', type='build')
+    depends_on('python@2.6.4:', type='build')
+    depends_on('py-mako@0.3.4:', type='build')
+    depends_on('gettext')
+    depends_on('icu4c')
+    depends_on('expat')
     depends_on('libpthread-stubs')
-    depends_on('libdrm')
     depends_on('openssl')
+    depends_on('xproto')
+    depends_on('glproto@1.4.14:')
+    depends_on('presentproto@1.0:')
     depends_on('libxcb@1.9.3:')
-    depends_on('libxshmfence@1.1:', when='~swrender')
-    depends_on('libx11', when='~swrender')
-    depends_on('libxext', when='~swrender')
-    depends_on('libxdamage', when='~swrender')
+    depends_on('libx11')
```

Comment:
Why is this no longer conditioned on ```swrender```? I'm primarily curious since it is a reasonably large library. Perhaps in your own testing it was required?

Reply:
libx11 is used to enable glx support, which we now provide in all mesa variants. If the libx11 dependency causes problems, an extra variant `~X` could be added to the package. But we have tried to keep the number of variants to a minimum, and libx11 is typically quick and easy to build in comparison with the other variants.

Reply:
Gotcha - thanks!